app/src/main/java/com/pvp/app/ui/screen/calendar/CalendarScreen.kt

```diff
@@ -291,6 +292,80 @@ enum class TaskFilter(val displayName: String) {
     Meal("Meal")
 }
 
+@Composable
+fun StepCounter(
+    model: CalendarViewModel = hiltViewModel(),
+    date: LocalDate
+) {
+    // Required for checking whether user has permissions before entering the window,
+    // as users can revoke permissions at any time
+    val permissionContract = PermissionController.createRequestPermissionResultContract()
+
+    val launcher =
+        rememberLauncherForActivityResult(permissionContract) {
+            model.getDaysSteps(date)
+        }
+
+    LaunchedEffect(Unit) {
+        if (model.permissionsGranted()) {
+            model.getDaysSteps(date)
+        } else {
+            launcher.launch(PERMISSIONS)
+        }
+    }
+
+    val steps = model.stepsCount.collectAsStateWithLifecycle()
+    val goal = 10000f // TODO create way for user to set a step goal?
+    val progress = steps.value / goal
+
+
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = Modifier.fillMaxSize()
+    ) {
+        // These have to be set here because for some reason MaterialTheme.colorScheme
+        // can't be used inside a Canvas ¯\_(ツ)_/¯
+        val backgroundArcColor = MaterialTheme.colorScheme.primaryContainer
+        val progressArcColor = MaterialTheme.colorScheme.primary
+
+        Canvas(modifier = Modifier.size(100.dp)) {
+            val strokeWidth = 6.dp.toPx()
+            val radius = 300f
+            val topLeft = Offset(
+                (size.width / 2) - (radius / 2),
+                (size.height / 2) - (radius / 2)
+            )
+            val size = Size(radius, radius)
+
+            drawArc(
+                color = backgroundArcColor,
+                startAngle = -90f,
+                sweepAngle = 360f,
+                useCenter = false,
+                topLeft = topLeft,
+                size = size,
+                style = Stroke(width = strokeWidth, cap = StrokeCap.Round)
```

Comment:
Multiple parameters could be split: here, below and in HealthConnectorServiceImpl. Be consistent.